tools/integration_tests/emulator_tests/proxy_server/main.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+var (
+	// Flag to accept config-file path.
+	fConfigPath = flag.String("config-path", "configs/config.yaml", "Path to the file")
+	// Flag to turn on debug logs.
+	debug = flag.Bool("debug", false, "logs will be enabled with a flag value of true.")
+	// Initialized before the server gets started.
+	gConfig *Config
+)
+
+// Host address of the proxy server.
+// TODO: Allow this value to be configured via a command-line flag.
+const PORT = "8020"
+
+type ProxyHandler struct {
+	http.Handler
+}
+
+func (ph ProxyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	targetURL := fmt.Sprintf("%s%s", gConfig.TargetHost, r.RequestURI)
+	req, err := http.NewRequest(r.Method, targetURL, r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	for name, values := range r.Header {
+		for _, value := range values {
+			req.Header.Add(name, value)
+		}
+	}
+
+	// TODO: Handler request and add retry-id according to the instruction.
```

Comment:
Do we need this TODO?

Reply:
I will add request handler in upcoming PR to add ID in request header, that's why I added TODO.